sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -246,6 +246,9 @@ private[hive] trait HiveInspectors {
    * Wraps with Hive types based on object inspector.
    */
   protected def wrapperFor(oi: ObjectInspector, dataType: DataType): Any => Any = oi match {
+    case _ if dataType.isInstanceOf[UserDefinedType[_]] =>
```

Comment:
- This codepath is shared by many things apart from ORC. Won't those be affected ?
- I would put this case in the every end. The reason being `UserDefinedType` are not that common compared to other types (esp. primitive types). So putting it below in the switch case will be better for perf.